src/shared/types/api.ts

```diff
@@ -5,6 +5,23 @@ export type IpcEventHandler<R = any> = (
   ...args: any[]
 ) => Promise<R>;
 
+export interface DirectoryEntryBase {
+  name: string;
+  details?: {
+    dateModifiedMs: number;
+  };
+}
+
+export interface FolderInfo extends DirectoryEntryBase {
+  directory: { isDirectory: true; content: DirectoryEntryInfo[] | undefined };
+}
+
+export interface FileInfo extends DirectoryEntryBase {
+  directory: { isDirectory: false; content: null };
+}
```

Comment:
Any reason for the missing content to once be `null` and once `undefined`?

Reply:
My reasoning was, that `null` means explicit lack of value (as non-directory cant have content in this case) and `undefined` means lack of value for unspecified reasons (in this case, value not yet assigned as data is not loaded yet), but maybe it's too confusing? My first implementation had simply `{ isDirectory: false }` for files, but having the same shape for both file and folder was simpler to deal with.

Reply:
I guess it could be simplified to just being `{ isDirectory: true; content: DirectoryEntryInfo[] | null }` for both file and folder, as: 
 - If `isDirectory` is false, it is a file and explicitely has no content
 - If `isDirectory` is true and `content` is null, it is a directory and content (explicitely) has not been loaded yet
 - If the directory is empty, `content` should be an empty array

What do you think?

Reply:
@LostInStatic I like this approach. Empty array for empty directory is also more controllable since you will not need to perform checks before iterating on the array. In terms of null vs undefined it's always a matter of preference due the behaviors of both . Some will only use null and some only undefined. I'd say just pick one for this type. Using both null and undefined in the same type could get confusing and cause weird conditional checks sometimes.